app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -381,7 +381,14 @@ fun CommentSection(
     Text(text = "Comments", style = MaterialTheme.typography.headlineSmall)
 
     // Display all comments
-    comments.forEach { comment -> CommentItem(profileId, comment, onReplyComment, onDeleteComment) }
+    comments.forEach { comment ->
+      CommentItem(
+          profileId,
+          comment,
```

Comment:
The update to the CommentItem constructor call to include the boolean allowReplies is a great enhancement. This change effectively allows to enable or disable replies for all top-level comments depending on the situation, which can significantly improve user interaction within the application.